src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -596,6 +641,26 @@ public static void CopyConstructor_CopiesAllPublicProperties()
             VerifyOptionsEqual(options, newOptions);
         }
 
+        [Fact]
+        public static void CopyConstructor_CopiesJsonSerializerContext()
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions();
+            options.AddContext<JsonContext>();
+            JsonContext original = Assert.IsType<JsonContext>(options.TypeInfoResolver);
+
+            // copy constructor copies the JsonSerializerContext
+            var newOptions = new JsonSerializerOptions(options);
+            Assert.Same(original, newOptions.TypeInfoResolver);
+
+            // resolving metadata returns metadata tied to the new options
+            JsonTypeInfo typeInfo = newOptions.TypeInfoResolver.GetTypeInfo(typeof(int), newOptions);
+            Assert.Same(typeInfo.Options, newOptions);
+
+            // it is possible to reset the resolver
+            newOptions.TypeInfoResolver = null;
+            Assert.IsType<DefaultJsonTypeInfoResolver>(newOptions.TypeInfoResolver);
+        }
+
         [Fact]
         public static void CopyConstructor_NullInput()
```

Comment:
I was expecting some more tests would have to be adjusted but that doesn't seem to be the case, I'm sure there was some coverage there, are we still doing fallback reflection in some cases or really no coverage?